app/components/channel_list/categories/header/header.test.tsx

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Database, Q} from '@nozbe/watermelondb';
+import React from 'react';
+
+import {MM_TABLES} from '@constants/database';
+import {renderWithIntlAndTheme} from '@test/intl-test-helper';
+import TestHelper from '@test/test_helper';
+
+import CategoryHeader from './header';
+
+import type CategoryModel from '@typings/database/models/servers/category';
+
+describe('components/channel_list/categories/header', () => {
+    let database: Database;
+    let category: CategoryModel;
+    beforeAll(async () => {
+        const server = await TestHelper.setupServerDatabase();
+
+        database = server.database;
+        const categories = await database.get<CategoryModel>(MM_TABLES.SERVER.CATEGORY).query(
+            Q.take(1),
+        ).fetch();
+
+        category = categories[0];
+    });
+
+    it('should match snapshot', () => {
```

Comment:
Nit: We could add another test for the other case (no channels and it is the favorites category). If we want to be really thorough, we could also add a test for other category without channels (not sure if it can happen), and favorites with some channels.